code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -485,69 +490,75 @@
-			if(tank_air.return_temperature() > 0)
-				transfer_moles = pressure_delta*cabin_air.return_volume()/(cabin_air.return_temperature() * R_IDEAL_GAS_EQUATION)
-				var/datum/gas_mixture/removed = tank_air.remove(transfer_moles)
-				cabin_air.merge(removed)
-		else if(pressure_delta < 0) //cabin pressure higher than release pressure
-			var/datum/gas_mixture/t_air = return_air()
-			pressure_delta = cabin_pressure - release_pressure
-			if(t_air)
-				pressure_delta = min(cabin_pressure - t_air.return_pressure(), pressure_delta)
-			if(pressure_delta > 0) //if location pressure is lower than cabin pressure
-				transfer_moles = pressure_delta*cabin_air.return_volume()/(cabin_air.return_temperature() * R_IDEAL_GAS_EQUATION)
-				var/datum/gas_mixture/removed = cabin_air.remove(transfer_moles)
-				if(t_air)
-					t_air.merge(removed)
-				else //just delete the cabin gas, we're in space or some shit
-					qdel(removed)
+		process_internal_damage_effects(seconds_per_tick)
+	if(cabin_sealed)
+		process_cabin_air(seconds_per_tick)
+	if(length(occupants))
+		process_occupants(seconds_per_tick)
+	if(mecha_flags & LIGHTS_ON)
+		use_power(light_energy_drain * seconds_per_tick)
 
+/obj/vehicle/sealed/mecha/proc/process_overclock_effects(seconds_per_tick)
+	if(!overclock_mode && overclock_temp > 0)
+		overclock_temp -= seconds_per_tick
+		return
+	overclock_temp = min(overclock_temp + seconds_per_tick, overclock_temp_danger * 2)
+	if(overclock_temp < overclock_temp_danger)
+		return
+	var/damage_chance = 100 * ((overclock_temp - overclock_temp_danger) / (overclock_temp_danger * 2))
+	if(SPT_PROB(damage_chance, seconds_per_tick))
+		do_sparks(5, TRUE, src)
+		try_deal_internal_damage(damage_chance)
+		take_damage(seconds_per_tick, BURN, 0, 0)
+
+/obj/vehicle/sealed/mecha/proc/process_internal_damage_effects(seconds_per_tick)
+	if(internal_damage & MECHA_INT_FIRE)
+		if(!(internal_damage & MECHA_INT_TEMP_CONTROL) && SPT_PROB(2.5, seconds_per_tick))
+			clear_internal_damage(MECHA_INT_FIRE)
+		if(cabin_air && cabin_sealed && cabin_air.return_volume()>0)
+			if(cabin_air.return_pressure() > (PUMP_DEFAULT_PRESSURE * 30) && !(internal_damage & MECHA_CABIN_AIR_BREACH))
+				set_internal_damage(MECHA_CABIN_AIR_BREACH)
+			cabin_air.temperature = min(6000+T0C, cabin_air.temperature+rand(5,7.5)*seconds_per_tick)
+			if(cabin_air.return_temperature() > max_temperature/2)
+				take_damage(seconds_per_tick*2/round(max_temperature/cabin_air.return_temperature(),0.1), BURN, 0, 0)
+
+	if(internal_damage & MECHA_CABIN_AIR_BREACH && cabin_air && cabin_sealed) //remove some air from cabin_air
+		var/datum/gas_mixture/leaked_gas = cabin_air.remove_ratio(SPT_PROB_RATE(0.05, seconds_per_tick))
+		if(loc)
+			loc.assume_air(leaked_gas)
+		else
+			qdel(leaked_gas)
+
+	if(internal_damage & MECHA_INT_SHORT_CIRCUIT && get_charge())
+		spark_system.start()
+		use_power(min(10 * seconds_per_tick, cell.charge))
+		cell.maxcharge -= min(10 * seconds_per_tick, cell.maxcharge)
```

Comment:
you shouldn't be changing the cells `maxcharge` this value is constant based on the cell type. Instead call `/obj/item/stock_parts/cell/use()` It does blackbox recording and also checks if the cell is rigged

https://github.com/tgstation/tgstation/blob/f9ca87b95976476540f70128809ad837967d9f6b/code/modules/power/cell.dm#L141-L150

Reply:
Check elsewhere and make the same changes

Reply:
It's the previous behavior of this type of internal damage. I assume the author wanted to make the cell capacity to decay on purpose.

Reply:
Actually never mind. `use_power()` already calls this proc. But i'm not sure why the cells max charge should be decreased over time. I guess that's a feature.

Reply:
Yeah you right. You can resolve this

Reply:
I added the maxcharge display to the UI so that the pilot can see the battery degrading over time. It was unclear when the power was going down together with maxcharge and percentage barely changed.